docs/source/tutorial.rst

```diff
@@ -0,0 +1,407 @@
+
+Over the domain :math:`[0,1]^{10}`, the integral of ``f_2`` is
+:math:`\int_{0}^{1} \dotsc \int_{0}^{1} \sum_{i=1}^{10} \sin(x_{i}) = 20 \sin^{2}(1/2) = 4.59697694131860282599063392557 \dotsc`
+
+Plotting this is tricky, so let’s directly move to the integrals.
+
+.. code:: ipython3
+
+    def f_2(x):
+        return torch.sum(torch.sin(x),dim=1)
+    
+    solution = 20*(torch.sin(torch.tensor([0.5]))*torch.sin(torch.tensor([0.5])))
+
+Let’s start with just 3 points per dimension, i.e., :math:`3^{10}=59,049` sample points. 
+
+**Note**: *torchquad* currently only supports equal numbers of points per dimension. 
+We are working on giving the user more flexibility on this point.
+
+.. code:: ipython3
+
+    integration_domain = [[0, 1]]*10 # Integration domain always is a list of lists to allow arbitrary dimensionality
+    N = 3**10 
+
+.. code:: ipython3
+
+    tp = Trapezoid()  # Initialize a trapezoid solver
+    result = tp.integrate(f_2, dim=10, N=N, integration_domain=integration_domain)
+    print_error(result,solution)
+
+
+.. parsed-literal::
+
+    **Output:** Results: 4.500804901123047
+            Abs. Error: 9.61723328e-02
+            Rel. Error: 2.09207758e-02
+    
+
+.. code:: ipython3
+
+    simp = Simpson()  # Initialize Simpson solver
+    result = simp.integrate(f_2, dim=10, N=N, integration_domain=integration_domain)
+    print_error(result,solution)
+
+
+.. parsed-literal::
+
+    **Output:** Results: 4.598623752593994
+            Abs. Error: 1.64651871e-03
+            Rel. Error: 3.58174206e-04
+    
+
+.. code:: ipython3
+
+    mc = MonteCarlo()
+    result = mc.integrate(f_2, dim=10, N=N, integration_domain=integration_domain, seed=42)
+    print_error(result,solution)
+
+
+.. parsed-literal::
+
+    **Output:** Results: 4.598303318023682
+            Abs. Error: 1.32608414e-03
+            Rel. Error: 2.88468727e-04
+    
+
+.. code:: ipython3
+
+    vegas = VEGAS()
+    result = vegas.integrate(f_2,dim=10,N=N,integration_domain=integration_domain)
+    print_error(result,solution)
+
+
+.. parsed-literal::
+
+    **Output:** Results: 4.598696708679199
+            Abs. Error: 1.71947479e-03
+            Rel. Error: 3.74044670e-04
+    
+
+Note that the Monte Carlo methods are much more competitive for
+this case. The bad convergence properties of the trapezoid method are
+visible while Simpson’s rule is still OK given the comparatively smooth
+integrand.
+
+If you have been repeating the examples from this tutorial on your own computer, you could also try 
+increasing N to :math:`5^{10}=9,765,625`.
+You can see the curse of dimensionality fully at play here, and 
+some users might even experience running out of memory at this point.
+
+Comparison with scipy
+---------------------
+
+Let’s explore how *torchquad*’s performance compares to scipy, the go-to
+tool for numerical integration. A more detailed exploration of this
+topic can be found in the *Benchmark_notebook.ipynb*. For simplicity,
```

Comment:
I think the Benchmark notebook  will be removed as well. So, I will avoid to point to it for more info.

Reply:
Good point